config/src/generator.rs

```diff
@@ -33,12 +33,8 @@ pub fn validator_swarm(
         nodes.push(node);
     }
 
-    let mut seed_peers = SeedPeersConfig::default();
-    let network = nodes[0].validator_network.as_ref().unwrap();
-    seed_peers
-        .seed_peers
-        .insert(network.peer_id, vec![network.listen_address.clone()]);
-
+    // set the first validator as every validators' initial configured seed peer.
+    let seed_peers = build_seed_peers(&nodes[0]);
```

Comment:
swarm is only used in tests right?

Reply:
right

Reply:
just a general feedback, but reviewing these parts it's not often super clear that they're test only. A combination of no comments indicating this, and no `#[cfg(test)]` annotation. Maybe worth thinking about that, and maybe even moving some of the code to testsuite if it's possible.